File: backend/src/app/file_diff/read_file_diff.rs
use std::path::Path;

use crate::app::file_diff::contracts::FileDiffReader;
use crate::domain::errors::AppError;

pub fn read_file_diff(
    file_diff_reader: &dyn FileDiffReader,
    repository_path: &Path,
    commit_id: &str,
    file_path: &str,
) -> Result<String, AppError> {
    file_diff_reader.read_file_diff(repository_path, commit_id, file_path)
}

File: frontend/src/modules/graph/application/use-cases/readFileDiff.ts
import { FileDiffReader } from "@modules/graph/application/contracts/FileDiffReader";

export async function readFileDiff(
  fileDiffReader: FileDiffReader,
  repositoryPath: string,
  commitId: string,
  filePath: string,
  staged?: boolean
): Promise<string> {
  return fileDiffReader.readFileDiff(repositoryPath, commitId, filePath, staged);
}
